algorithms-and-data-structures-part-2/index.ts

```diff
@@ -0,0 +1,58 @@
+const PRIORITY = 100;
+const MAX_JOBS = 10000;
+
+class JobRunner {
+    private jobs: Job[] = [];
+    private priorityMap: Map<number, Job[]> = new Map();
+    
+    constructor() {
+        for (let i = 0; i < MAX_JOBS; i++) {
+            this.jobs.push(new Job());
+        }
+    }
+
+    sortJobsByPriority() {
+        this.jobs.forEach((job) => {
+            if (this.priorityMap.has(job.priority)) {
+                this.priorityMap.get(job.priority)!.push(job);
+            } else {
+                this.priorityMap.set(job.priority, [job]);
+            }
+        });
+    }
+
+    sortMapByPriority() {
+        const sortedMap = new Map([...this.priorityMap.entries()].sort((a, b) => b[0] - a[0]));
+        this.priorityMap = sortedMap;
+    }
+    
+    run() {
+        console.time();
+        this.sortJobsByPriority();
```

Comment:
ideally is to use max priority queue data structure - then you will always have the most prioritized element on the top of the tree. Give even more benefits when you continue adding jobs while executing them.